src/visualizers/FactorFence.ts

```diff
@@ -0,0 +1,738 @@
+        // there's a bug here that when the keyboard is
+        // pressed in this window but
+        // released in another window, keyIsPressed stays
+        // positive and the sketch keeps computing,
+        // e.g. if you tab away from the window
+        if (this.sketch.keyIsPressed) {
+            this.keyPresses()
+        } else {
+            ++this.frame
+        }
+
+        // clear the sketch
+        this.sketch.clear(0, 0, 0, 0)
+        this.sketch.background(this.palette.backgroundColor)
+
+        // determine which terms will be on the screen so we only
+        // bother with those
+        // resets this.scaleFactor
+        const barsInfo = this.barsShowing()
+
+        // this scales the whole sketch
+        // must compensate when using invariant sketch elements
+        // like text
+        this.sketch.scale(this.scaleFactor)
+
+        this.mouseOn = false // flag mouse is not on graph by default
+
+        // try again if need more terms from cache
+        if (this.collectFailed) this.collectDataForScale(barsInfo)
+
+        // set factoring needed pointer
+        this.firstFailure = this.storeFactors(barsInfo)
+
+        // loop through the terms of the seq and draw the bars for each
+        for (
+            let myIndex = barsInfo.minBars;
+            myIndex < barsInfo.maxBars;
+            myIndex++
+        )
+            // note that this also watches for mouseover as each bar
+            // is drawn
+            this.drawTerm(myIndex)
+
+        // text at base of sketch, if not small canvas
+        if (this.sketch.height > 400 && this.labels) this.bottomText()
+
+        // stop drawing if no input from user
+        if (this.frame > 3) this.sketch.noLoop()
+    }
+
+    drawTerm(myIndex: number) {
+        // This function draws the full stacked bars for a single term
+        // Input is index of the term
+
+        // set colours
+        let bottomColor = this.palette.gradientBar.bottom
+        let topColor = this.palette.gradientBar.top
+
+        // get sign of term
+        let mySign = 1
+        if (this.seq.getElement(myIndex) < 0 && this.signs) mySign = -1
+
+        // get factors of term
+        const factors = this.factorizations[myIndex]
+
+        // we are drawing several bars
+        // on top of each other,
+        // so this height counter moves up as we go
+        let cumulHt = 0
+
+        for (const primeIsHigh of [true, false]) {
+            // first we do this with primeIsHigh = true
+            // (that means highlighted prime is at hand)
+            // then with primeIsHigh = false
+            // in this way the highlighted prime sits
+            // at the bottom of the stack
+
+            // loop through factor to draw for this term
+            // from smaller to larger
+            for (let facIndex = 0; facIndex < factors.length; facIndex++) {
+                // get the next prime factor
+                const factor = factors[facIndex]
+
```

Comment:
As far as I can see, you're not using facIndex anywhere in this loop below here. In that case, just iterate `for (factor of factors) {` ... then you don't need facIndex identifier at all and makes your intention clearer.

Reply:
got it

Reply:
committed